### ISSUE TITLE
[Bug]: zsh command not found

### ISSUE THREAD

**Issue by JameInw** [wavetermdev/waveterm#1511]
### Current Behavior

I tried writing Custom Widgets that are CLI Widgets
```json
"WakeOnLan-XXXXX": {
    "icon": "power-off",
    "label": "WoL-HA",
    "color": "#F72C5B",
    "blockdef": {
      "meta": {
        "view": "term",
        "controller": "cmd",
        "cmd": "wakeonlan FF:FF:FF:FF:FF:FF",
        "cmd:closeonexit" : 0
      }
    }
  }
```

But when I tried it, the result was

zsh:1: command not found: wakeonlan
process finished with exit code = 127

Even though I tried the command myself, it worked.
Note: I tested it on Macbook Air M2

### Expected Behavior

I hope it will work in the future.

### Steps To Reproduce

1. Macbook Air M2
2. Use ZSH
3. Use Script
```json
"WakeOnLan-XXXX": {
    "icon": "power-off",
    "label": "WoL",
    "color": "#F72C5B",
    "blockdef": {
      "meta": {
        "view": "term",
        "controller": "cmd",
        "cmd": "wakeonlan",
        "cmd:closeonexit" : 0
      }
    }
  }
```

### Wave Version

Client Version 0.10.1 (202412130327)

### Platform

macOS

### OS Version/Distribution

MacOS 15.1.1

### Architecture

arm64

### Questionnaire

- [ ] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [X] I don't have time to fix this right now, but maybe later

**Comment by esimkowitz** [comment]
To confirm, is ZSH your primary shell? Is this a custom script you've written?

**Comment by esimkowitz** [comment]
Ah ok we run custom widgets in the login shell to reduce overhead. This likely doesn't have your RC files loaded. As a workaround can you try the following:

```sh
source ~/.zshrc && wakeonlan
```

**Comment by JameInw** [comment]
> To confirm, is ZSH your primary shell? Is this a custom script you've written or something you've installed?

Yes, I mainly use ZSH shell, and yes, this is a script I wrote myself.

**Comment by JameInw** [comment]
> Ah ok we run custom widgets in the login shell to reduce overhead. This likely doesn't have your RC files loaded. As a workaround can you try the following:
> 
> ```shell
> source ~/.zshrc && wakeonlan
> ```

I was using the source ~/.zshrc command before using wavetrem.

**Comment by esimkowitz** [comment]
You shouldn't need to do that we source it for the terminal blocks, just for the command widget config

**Comment by rjousse18** [comment]
Hi ! Same here in Ubuntu 24.04 snap version

**Comment by mrztti** [comment]
Hey, I have had the same problem, only workaround I found is using this bypass

```json
"blockdef": {
      "meta": {
        "view": "term",
        "controller": "cmd",
        "cmd:shell": true,
        "cmd:interactive": true,
        "cmd:cwd": "/Users/...",
        "cmd": "tmpdir=$(mktemp -d) && echo 'source ~/.zshrc\nconda activate myenv\nexec ~/someinteractivescript.sh' > $tmpdir/.zshrc && ZDOTDIR=$tmpdir zsh -i"
      }
    }
```

This creates a temporary RC file which sources the real RC plus any command which needs it (like in my case conda and some interactive ssh script). The key is the cmd:shell property I believe.

Would be nice to have some more documentation on this!
